src/test/java/apoc/atomic/AtomicTest.java

```diff
@@ -0,0 +1,132 @@
+		Relationship rel = (Relationship) db.execute("MATCH (n:Person {name:'Angelo'})-[r:WORKS_FOR]-(c) return r;").next().get("r");
+		testCall(db, "CALL apoc.atomic.concat({rel},{property},{value})",map("rel",rel,"property","role","value","eloper"), (r) -> {});
+		assertEquals("software developer", db.execute("MATCH (n:Person {name:'Angelo'})-[r:WORKS_FOR]-(c) return r.role as role;").next().get("role"));
+	}
+
+	@Test
+	public void testRemoveArrayValueLong(){
+		db.execute("CREATE (p:Person {name:'Tom',age: [40,50,60]}) CREATE (c:Person {name:'John',age: 40}) CREATE (a:Person {name:'Anne',age: 22})");
+		Node node = (Node) db.execute("MATCH (n:Person {name:'Tom'}) return n;").next().get("n");
+		testCall(db, "CALL apoc.atomic.remove({node},{property},{position})",map("node",node,"property","age","position",1), (r) -> {});
+		assertEquals(Arrays.asList(40L, 60L).toArray(), new ArrayBackedList(db.execute("MATCH (n:Person {name:'Tom'}) return n.age as age;").next().get("age")).toArray());
+	}
+
+	@Test
+	public void testInsertArrayValueLong(){
+		db.execute("CREATE (p:Person {name:'Tom',age: 40}) CREATE (c:Person {name:'John',age: 40}) CREATE (a:Person {name:'Anne',age: 22})");
+		Node node = (Node) db.execute("MATCH (n:Person {name:'Tom'}) return n;").next().get("n");
+		testCall(db, "CALL apoc.atomic.insert({node},{property},{position},{value})",map("node",node,"property","age","position",2,"value",55L), (r) -> {});
+		assertEquals(Arrays.asList(40L,55L).toArray(), new ArrayBackedList(db.execute("MATCH (n:Person {name:'Tom'}) return n.age as age;").next().get("age")).toArray());
+	}
+
+	@Test
+	public void testInsertArrayValueLongRelationship(){
+		db.execute("CREATE (p:Person {name:'Tom',age: 40}) CREATE (c:Person {name:'John',age: 40}) CREATE (p)-[:KNOWS{since:[40,50,60]}]->(c)");
+		Relationship rel = (Relationship) db.execute("MATCH (n:Person {name:'Tom'})-[r:KNOWS]-(c) return r;").next().get("r");
+		testCall(db, "CALL apoc.atomic.insert({rel},{property},{position},{value})",map("rel",rel,"property","since","position",2,"value",55L), (r) -> {});
+		assertEquals(Arrays.asList(40L, 50L, 55L, 60L).toArray(), new ArrayBackedList(db.execute("MATCH (n:Person {name:'Tom'})-[r:KNOWS]-(c) return r.since as since;").next().get("since")).toArray());
+	}
+
+	@Test
+	public void testUpdateNode(){
+		db.execute("CREATE (p:Person {name:'Tom',salary1: 1800, salary2:1500})");
+		Node node = (Node) db.execute("MATCH (n:Person {name:'Tom'}) return n;").next().get("n");
+		testCall(db, "CALL apoc.atomic.update({node},{property},{operation})",map("node",node,"property","salary1","operation","n.salary1 + n.salary2"), (r) -> {});
+		assertEquals(3300L, db.execute("MATCH (n:Person {name:'Tom'}) return n.salary1 as salary;").next().get("salary"));
+	}
+
+	@Test
+	public void testUpdateRel(){
+		db.execute("CREATE (t:Person {name:'Tom'})-[:KNOWS {forYears:5}]->(m:Person {name:'Mary'})");
```

Comment:
you can just return the relationship from here
or just test the proc within the procedure.